### PR TITLE
Use Symfony Messenger for Search Backend Data

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves  partially https://github.com/pimcore/pimcore/issues/6371

### CODE REVIEW

File: bundles/CoreBundle/EventListener/SearchBackendListener.php [hunk 1]
```diff
@@ -52,8 +54,10 @@ public static function getSubscribedEvents()
      */
     public function onPostAddElement(ElementEventInterface $e)
     {
-        $searchEntry = new Data($e->getElement());
-        $searchEntry->save();
+        $element = $e->getElement();
+        \Pimcore::getContainer()->get(MessageBusInterface::class)->dispatch(
```
Comment (dpfaffenbauer) [inline bundles/CoreBundle/EventListener/SearchBackendListener.php]: you are in an service here and still use Pimcore::getContainer(). Use DI instead.

File: bundles/CoreBundle/EventListener/SearchBackendListener.php [hunk 2]
```diff
@@ -73,12 +77,8 @@ public function onPreDeleteElement(ElementEventInterface $e)
...
     {
         $element = $e->getElement();
-        $searchEntry = Data::getForElement($element);
-        if ($searchEntry instanceof Data && $searchEntry->getId() instanceof Data\Id) {
-            $searchEntry->setDataFromElement($element);
-            $searchEntry->save();
-        } else {
-            $this->onPostAddElement($e);
-        }
+        \Pimcore::getContainer()->get(MessageBusInterface::class)->dispatch(
```
Comment (dpfaffenbauer) [inline bundles/CoreBundle/EventListener/SearchBackendListener.php]: you are in an service here and still use Pimcore::getContainer(). Use DI instead.

**Review by dpfaffenbauer — CHANGES_REQUESTED** [review]

File: bundles/CoreBundle/EventListener/SearchBackendListener.php (hunk 1 above)
Comment (dvesh3) [reply bundles/CoreBundle/EventListener/SearchBackendListener.php]: done

File: bundles/CoreBundle/EventListener/SearchBackendListener.php (hunk 2 above)
Comment (dvesh3) [reply bundles/CoreBundle/EventListener/SearchBackendListener.php]: done

**Pull request merged**